### PR TITLE
build(deps): update github.com/docker/distribution to fix OCI Manifest Type Confusion Issue

### PR DESCRIPTION
**Pull request by machi1990**
## Description

Details on https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/security/dependabot/2

## Verification Steps

See the proposed version on https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/security/dependabot/2

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -72,7 +72,7 @@ require (
 	github.com/cucumber/messages-go/v16 v16.0.1 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
-	github.com/docker/distribution v2.7.1+incompatible // indirect
+	github.com/docker/distribution v2.8.1+incompatible // indirect
```
Comment (machi1990) [inline go.mod:75]: We could consider https://github.com/distribution/distribution in the future as github.com/docker/distribution redirects to it.

**Review by miguelsorianod — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (miguelsorianod) [reply go.mod:75]: After reviewing it, I don't think we can't change that:

It is an indirect dependency, so the dependencies that use that dependency should change it to be propagated to our go mod/sum files

Comment (miguelsorianod) [reply go.mod:75]: Specifically:
`github.com/docker/go-healthcheck`

Comment (machi1990) [reply go.mod:75]: Ah thanks for checking. I thought we could be able to re-use it somehow.
You reckon that we are good to merge this at it's state? 

> Specifically: `github.com/docker/go-healthcheck`
> 
> and it is marked as readonly so it is unlikely to change

Also this package is archived, I am wondering if there are better alternatives out there.

Comment (miguelsorianod) [reply go.mod:75]: >  You reckon that we are good to merge this at it's state?

Let's merge it

**Pull request merged**